registry/validate_doc_update.js

```diff
@@ -371,10 +371,6 @@ module.exports = function (doc, oldDoc, user, dbCtx) {
 
   function checkDep(version, dep, t) {
     ridiculousDeps()
-    if (!entity) {
```

Comment:
so based on the code above `entity` is parsed from package name (as the 1st part). e.g: `@foo/bar` makes `entity = foo`. 

This check is basically if I'm publishing non-scoped packages (considered global, like `request`, or `webpack`), it cannot depends on scoped packages (which is considered private at the time). But if I'm publishing `@foo/bar`, then it can depends on private packages because `@foo/bar` is considered private.